enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/transaction/log/checkpoint/ConfigurableIOLimiter.java

```diff
@@ -116,13 +158,31 @@ public long maybeLimitIO( long previousStamp, int recentlyCompletedIOs, Flushabl
     @Override
     public void disableLimit()
     {
-        disableCountUpdater.getAndIncrement( this );
+        long currentState;
+        long newState;
+        do
+        {
+            currentState = stateUpdater.get( this );
```

Comment:
That is by design, actually. This call is used to temporarily disable the limit for the purpose of speeding up flushing. For instance, copying a store over the network requires a recent check-point, which means that the check-point process now suddenly comes under pressure to complete as soon as possible. If, say, we have both an online back-up being taken, and a new member joins the cluster, then two store copy processes will be started and they will both call `disableLimit`, and when they have their check-point they will both `enableLimit`. This way, the check-pointer won't slow down again until the last urgent need for a check-point has been serviced.

It might not be obvious from the method name, but the javadocs explains it.

Reply:
Makes sense thx!